test/test_hessian.jl

```diff
@@ -1,12 +1,17 @@
 using SparseConnectivityTracer
-using SparseConnectivityTracer: HessianTracer, MissingPrimalError, tracer, trace_input
+using SparseConnectivityTracer:
+    Dual, HessianTracer, MissingPrimalError, tracer, trace_input, empty
 using SparseConnectivityTracer: DuplicateVector, RecursiveSet, SortedVector
 using ADTypes: hessian_sparsity
 using SpecialFunctions: erf, beta
 using Test
 
-const SECOND_ORDER_SET_TYPES = (
-    BitSet, Set{UInt64}, DuplicateVector{UInt64}, RecursiveSet{UInt64}, SortedVector{UInt64}
+SECOND_ORDER_SET_TYPE_TUPLES = (
+    (BitSet, Set{Tuple{Int,Int}}),  #
+    (Set{UInt64}, Set{Tuple{UInt64,UInt64}}),
+    (DuplicateVector{UInt64}, DuplicateVector{Tuple{UInt64,UInt64}}),
+    # (RecursiveSet{UInt64}, RecursiveSet{Tuple{UInt64,UInt64}}),  # TODO: code union!
+    # (SortedVector{UInt64}, SortedVector{Tuple{UInt64,UInt64}}),  # TODO: fix bug
 )
```

Comment:
Would be be great to run a subset of the tests over combinations of "Gradient Set" and "Hessian Set" types using nested loops.

Reply:
done

Reply:
Not really, because it makes no sense to want a `HessianTracer{DuplicateVector{UInt64}, SortedVector{Tuple{UInt64,UInt64}}}`. Not all combinations are interesting

Reply:
I was never talking about all combinations. ;)

Reply:
The most efficient set type for `Int` isn't *necessarily* the most efficient set type on `Tuple{Int, Int}` as well.

Adding these tests would make sure that enough non-leaky abstractions are in place for `GradientTracer` and `HessianTracer` to interact.

Reply:
> Not all combinations are interesting

In other words: it's a test for code quality, not for performance.